Azure.ResourceManager.Core/ResourceGroupOperations.cs

```diff
@@ -30,8 +31,15 @@ public class ResourceGroupOperations : ResourceOperationsBase<ResourceGroup>,
         internal ResourceGroupOperations(SubscriptionOperations options, string rgName)
             : base(options, $"{options.Id}/resourceGroups/{rgName}")
         {
+            if (rgName.Length > 90)
+                throw new ArgumentOutOfRangeException("ResourceGroupName cannot be longer than 90 characters.");
+
+            if (!ValidationPattern.IsMatch(rgName))
+                throw new ArgumentException("ResourceGroupName cannot have a special character.");
```

Comment:
we might just print the regex here instead of saying special characters because it can't contain whitespace either